mountpoint-s3/tests/common/cache.rs

```diff
@@ -0,0 +1,126 @@
+use std::{
+    sync::{
+        atomic::{AtomicU64, Ordering},
+        Arc,
+    },
+    time::Duration,
+};
+
+use async_trait::async_trait;
+use mountpoint_s3::{
+    data_cache::{BlockIndex, ChecksummedBytes, DataCache, DataCacheResult},
+    object::ObjectId,
+};
+
+/// A wrapper around any type implementing [DataCache], which counts operations
+pub struct CacheTestWrapper<Cache> {
```

Comment:
Is this a test wrapper, or a metrics wrapper? It feels to me this would work fine for emitting metrics for the most part